### ISSUE TITLE
Kubelet Service created by Prometheus Operator is not deleted when uninstalling chart

### ISSUE THREAD

**Issue by frankreno** [SumoLogic/sumologic-kubernetes-collection#1101]
When you install our collection, a service to collect kubelet metrics is created by the operator (eg direct creation by operator and not by template).  However, when you uninstall the chart, the operator does not clean this up.

The issue is the dangling service can be picked up by a future install.  Since the service is named based on the release name given at the time of install, you can have 2 different kubelet services.  The serviceMonitor sees them both and this can lead to duplicated metrics ingestion for Kubelet.

https://github.com/helm/charts/issues/18761

**Comment by swiatekm-sumo** [comment]
A quick summary of my findings:

It is possible to delete the Service in a Helm post-delete hook, if a bit awkward. Doing so would require either really ugly hacks in our chart, or making the change upstream in kube-prometheus-stack.

It's also possible to tell prometheus-operator not to create the Service in the first place, and the kube-prometheus-stack chart exposes this option. We could then create the Service manually and delete it like any other resource on uninstall. Unfortunately, the upstream chart also does other things depending on this option - in this case, [alerts for KubeletDown](https://github.com/prometheus-community/helm-charts/blob/5477284197a4075229e80decbb89ebb963d9eb15/charts/kube-prometheus-stack/templates/prometheus/rules-1.14/kubernetes-system-kubelet.yaml#L175). These are easy to reproduce right now, but we'd be introducing an additional maintenance burden if we'd wanted to keep our chart's behaviour consistent with upstream's.

It's doesn't appear possible to make the ServiceMonitor only target a specific kubelet Service. ServiceMonitors can only be targeted based on namespace and labels, and the labels for this Service are [hardcoded](https://github.com/prometheus-operator/prometheus-operator/blob/33694a16bee02c50b9468f47d4a62af90e6dad1f/pkg/prometheus/operator.go#L585) in prometheus-operator.

In summary, we have the following solutions available (these are not mutually exclusive):
* Try to solve the underlying issue in upstream by adding a post-delete hook to remove the Service.
* Disable automatic creation of the Service and create it manually in our chart, along with whatever else upstream creates based on that flag.
* Add this to our documentation under Known Issues and Uninstall instructions.

**Comment by swiatekm-sumo** [comment]
Opened an issue upstream: https://github.com/prometheus-community/helm-charts/issues/1523.

**Comment by swiatekm-sumo** [comment]
I'm going to close this issue here. We're not using prometheus-operator by default anymore, and we plan to get rid of it completely in v5.
